librz/analysis/dwarf_process.c

```diff
@@ -1700,16 +1713,22 @@ RZ_API void rz_analysis_dwarf_integrate_functions(RzAnalysis *analysis, RzFlag *
 				rz_flag_unset_off(flags, offset);
 				rz_flag_set_next(flags, global_name, offset, 4);
 				free(global_name);
-			} else if (*kind == 's' && fcn) {
-				rz_analysis_function_set_var(fcn, offset - fcn->maxstack, *kind, ttype, 4, false, var_name);
 			} else if (*kind == 'r' && fcn) {
 				RzRegItem *i = rz_reg_get(analysis->reg, extra, -1);
 				if (!i) {
 					goto loop_end;
 				}
-				rz_analysis_function_set_var(fcn, i->index, *kind, ttype, 4, false, var_name);
-			} else if (fcn) { /* kind == 'b' */
-				rz_analysis_function_set_var(fcn, offset - fcn->bp_off, *kind, ttype, 4, false, var_name);
+				RzAnalysisVarStorage stor;
+				rz_analysis_var_storage_init_reg(&stor, extra);
+				rz_analysis_function_set_var(fcn, &stor, ttype, 4, var_name);
+			} else if (fcn) { /* kind == 'b' || kind == 's' || kind == 'c' */
+				RzAnalysisVarStorage stor;
+				RzStackAddr addr = offset;
+				if (*kind == 'b') {
```

Comment:
What about using an enum here?

Reply:
Could be done, but this comes from sdb and should rather be refactored completely to not use sdb at all.